src/main/java/org/alfresco/service/cmr/repository/ContentService.java

```diff
@@ -154,12 +154,18 @@ public ContentWriter getWriter(NodeRef nodeRef, QName propertyQName, boolean upd
     public ContentWriter getTempWriter();
 
     /**
-     * Gets a presigned URL to directly access a binary content. It is up to the content store
-     * if it can fulfil this request with an expiry time or not.
+     * Gets a presigned URL to directly access a binary content. It is up to the
+     * content store if it can fulfil this request with an expiry time (in
+     * milliseconds) or not.
      *
-     * @param contentUrl A content store URL
-     * @param expiryTime Expiration time in milliseconds 
-     * @return A direct access URL for a binary content or empty string if not supported
+     * @param nodeRef
+     *            a reference to a node having a content property
+     * @param expiryTime
+     *            the expiration time in milliseconds of the direct access url. This
+     *            is the length of time in milliseconds that the link is valid for.
+     * @return A direct access URL for a binary content or empty string if not
+     *         supported
```

Comment:
or returns `null` if there is no binary content for the node

Reply:
The AWS API generates an URL even though there is no binary content on the given path.

Reply:
Our ContentService returns null though.

Reply:
You're right. I was thinking of binary content in s3.

Reply:
Change done, good catch.